### DISCUSSION TITLE
requiring multiple claims with multi OP setup

### DISCUSSION THREAD

**Discussion by rajeevn1** [OpenIDC/mod_auth_openidc#770] [Q&A]
I am using 2.4.10-1~bullseye+1 release from github with multiple providers (google, globus, gitlab).

The following works

\<RequireAll>
\# Require not claim email_verified:false
Require claim email~^(test1@gmail\.com|test2@gmail\.com)$
\</RequireAll>

but when I un-comment the claim for email_verified, the authentication always fails.  I have confirmed that email_verified claim, as provided by OP, is either set to true (google, gitlab), or is missing (globus).

The issue does not exist if I use any single provider setup.

I would appreciate any help in resolving the issue.

**Comment by zandbelt** [comment]
Can you provide server debug logs of both cases (for the same OP) so the authorization process can be analyzed?

**Comment by rajeevn1** [comment]
Log files attached: I have deleted some email regexps and replaced with email~...$

non-working log file:
[eb.txt](https://github.com/zmartzone/mod_auth_openidc/files/7916294/eb.txt)

working log file:
[eg.txt](https://github.com/zmartzone/mod_auth_openidc/files/7916295/eg.txt)

**Reply by zandbelt** [reply]
thanks, I am able to reproduce this; it appears that combining `<RequireAll>` with Discovery does not work: including a(ny) 2nd Require directive in such a clause makes it fail; I'll need to investigate more to find a fix, to be continued

**Reply by zandbelt** [reply]
2.4.11 addresses this
